src/test/testrunner.ts

```diff
@@ -164,6 +164,34 @@ export class JDExprEvaluator {
     }
 }
 
+export class JDCommandClosure {
+    private currentCmdStatus = JDCommandStatus.NotStarted
+    private commandTimeOut = 10000
+
+    constructor(private cmd: jdtest.CommandSpec) {
+
+    }
+
+    semantics() {
+        switch(cmdToTestFunction(this.cmd).id) {
+            case 'say':
+                break
+            case 'ask':
```

Comment:
IMO, define functions in a class and just look them up using some javascript magic + call function.

Reply:
Doing some design work on paper while "on vacation"...